make.py

```diff
@@ -139,8 +140,37 @@ def run(args: Union[str, List[str]], cd: Optional[PathLike] = None) -> None:
 def run_doc(args: Union[str, List[str]]) -> None:
     run(args, cd=FULL_DOC_DIR)
 
-
-@app.command()
+def emit_makefile():
+    with cd_context(PROJECT_ROOT):
+        with open('Makefile', 'w') as f:
+            f.write('# DO NOT EDIT BY HAND: is automatically re-generated by make.py\n')
```

Comment:
If we're going this route, it may make more sense to also paste above every directive inside the loop. That still might not be enough.